lib/logstash/outputs/elasticsearch_java.rb

```diff
@@ -379,27 +343,26 @@ def register
       @host = ["localhost"]
     end
 
-    client_settings.merge! setup_ssl()
-    client_settings.merge! setup_proxy()
-
     common_options = {
       :protocol => @protocol,
       :client_settings => client_settings
     }
 
-    common_options.merge! setup_basic_auth()
+    # Update API setup
+    update_options = {
+      :upsert => @upsert,
+      :doc_as_upsert => @doc_as_upsert
+    }
+    common_options.merge! update_options if @action == 'update'
 
     client_class = case @protocol
       when "transport"
         LogStash::Outputs::Elasticsearch::Protocols::TransportClient
       when "node"
         LogStash::Outputs::Elasticsearch::Protocols::NodeClient
-      when /http/
-        LogStash::Outputs::Elasticsearch::Protocols::HTTPClient
     end
 
     if @embedded
```

Comment:
@andrewvc Lets remove `@embedded` functionality. There is no reason to start a ES server on a LS server anymore. This is a source of security issues, high heap usage in a LS node, and was only used as a POC tool.